### PR TITLE
Run CBMC on EC2 runner in CI and guard by label

### PR DESCRIPTION
**Pull request by hanno-becker**
Fixes https://github.com/pq-code-package/mlkem-c-aarch64/issues/117

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -96,17 +97,26 @@ jobs:
...
+            ec2_ami: ubuntu-latest (custom AMI)
+            ec2_ami_id: ami-08ddb0acd99dc3d33
+            cbmc: 'true'
     name: ${{ matrix.target.name }}
     permissions:
       contents: 'read'
       id-token: 'write'
     uses: ./.github/workflows/ci_ec2_reusable.yml
-    if: github.repository_owner == 'pq-code-package' || github.ref == 'refs/heads/main'
+    if: github.repository_owner == 'pq-code-package' && (github.event.label.name == 'ci' || github.ref == 'refs/heads/main')
```
Comment (mkannwischer) [inline .github/workflows/ci.yml:115]: I don't think this corresponds to what the title of the PR says. 
This actually disables all CI for PRs on `AMD EPYC 4th gen (t3a)` and `Intel Xeon 4th gen (t3)` unless you add the `ci` label. Was that intentional? 
If it was not you will have to change it such that only the cbmc part depends on the `ci` label.

**Review by mkannwischer — CHANGES_REQUESTED** [review]
I am a little confused - see my comment.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (hanno-becker) [reply .github/workflows/ci.yml:115]: Fair point. I don't think we want EC2 runners to be triggered by remote PRs, which is what was the case previously and is fixed here. I will amend the PR title and commit message.

Comment (hanno-becker) [reply .github/workflows/ci.yml:115]: I don't know how to run the EC2 CI _always_ (without labels) on PRs from the main repository.

**Review by mkannwischer — APPROVED** [review]
Okay, makes sense. Thanks.

**Pull request merged**